site/app/models/Notification.php

```diff
@@ -0,0 +1,223 @@
+            $this->setCurrentUser($this->core->getUser()->getId());
+            $this->setComponent($details['component']);
+            switch ($this->getComponent()) {
+                case 'forum':
+                    $this->handleForum($details);
+                    break;
+                default:
+                    // Prevent notification to be pushed in database
+                    $this->setComponent("invalid");
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Handles notifications related to forum
+     *
+     * @param array $details
+     */
+    private function handleForum($details) {
+        switch ($details['type']) {
+            case 'new_announcement':
+                $this->actAsNewAnnouncementNotification($details['thread_id'], $details['thread_title']);
+                break;
+            case 'updated_announcement':
+                $this->actAsUpdatedAnnouncementNotification($details['thread_id'], $details['thread_title']);
+                break;
+            case 'reply':
+                $this->actAsForumReplyNotification($details['thread_id'], $details['post_content'], $details['reply_to']);
+                break;
+            case 'merge_thread':
+                $this->actAsForumMergeThreadNotification($details['parent_thread_id'],  $details['parent_thread_title'], $details['child_thread_title'], $details['reply_to']);
+                break;
+            case 'edited':
+                $this->actAsForumEditedNotification($details['thread_id'],  $details['post_content'], $details['reply_to']);
+                break;
+            case 'deleted':
+                $this->actAsForumDeletedNotification($details['thread_id'],  $details['post_content'], $details['reply_to']);
+                break;
+            case 'undeleted':
+                $this->actAsForumUndeletedNotification($details['thread_id'],  $details['post_content'], $details['reply_to']);
+                break;
+            default:
+                return;
+        }
+    }
+
+    private function actAsNewAnnouncementNotification($thread_id, $thread_title) {
+        $this->setNotifyMetadata(json_encode(array('component' => 'forum', 'page' => 'view_thread', 'thread_id' => $thread_id)));
+        $this->setNotifyContent("New Announcement: ".$thread_title);
+        $this->setNotifySource($this->getCurrentUser());
+        $this->setNotifyTarget(null);
+    }
+
+    private function actAsUpdatedAnnouncementNotification($thread_id, $thread_title) {
+        $this->setNotifyMetadata(json_encode(array('component' => 'forum', 'page' => 'view_thread', 'thread_id' => $thread_id)));
+        $this->setNotifyContent("Announcement: ".$thread_title);
+        $this->setNotifySource($this->getCurrentUser());
+        $this->setNotifyTarget(null);
+    }
+
+    private function actAsForumReplyNotification($thread_id, $post_content, $target) {
+        // TODO: Redirect to post itself
```

Comment:
In the URL you can simply put #<post_id> at the end, try using the search feature and inspecting element on a resulting post.